### PR TITLE
Fixed #30411 -- Improved formatting of text tracebacks in technical 500 templates.

### PR DESCRIPTION
**Pull request by jerrodmartin**
Fixes [30411](https://code.djangoproject.com/ticket/30411).

### CODE REVIEW

File: django/views/templates/technical_500.txt [hunk 1]
```diff
@@ -27,12 +27,11 @@ In template {{ template_info.name }}, error at line {{ template_info.line }}
...
 {% for source_line in template_info.source_lines %}{% if source_line.0 == template_info.line %}   {{ source_line.0 }} : {{ template_info.before }} {{ template_info.during }} {{ template_info.after }}{% else %}   {{ source_line.0 }} : {{ source_line.1 }}{% endif %}{% endfor %}{% endif %}{% if frames %}
 
-Traceback:
+Traceback (most recent call last):
 {% for frame in frames %}{% ifchanged frame.exc_cause %}{% if frame.exc_cause %}
 {% if frame.exc_cause_explicit %}The above exception ({{ frame.exc_cause }}) was the direct cause of the following exception:{% else %}During handling of the above exception ({{ frame.exc_cause }}), another exception occurred:{% endif %}
-{% endif %}{% endifchanged %}
-File "{{ frame.filename }}" in {{ frame.function }}
-{% if frame.context_line %}  {{ frame.lineno }}. {{ frame.context_line }}{% endif %}
+{% endif %}{% endifchanged %}  File "{{ frame.filename }}"{% if frame.context_line %}, line {{ frame.lineno }}, in {{ frame.function }}{% endif %}
```
Comment (felixxm) [inline django/views/templates/technical_500.txt]: `{% endif %}` should be moved before `, in {{ frame.function }}{` i.e.
```
{% endif %}{% endifchanged %}  File "{{ frame.filename }}"{% if frame.context_line %}, line {{ frame.lineno }}{% endif %}, in {{ frame.function }}
```

File: tests/view_tests/tests/test_debug.py [hunk 2]
```diff
@@ -417,7 +417,7 @@ def test_reporting_frames_without_source(self):
         html = reporter.get_traceback_html()
         self.assertIn('generated in funcName', html)
         text = reporter.get_traceback_text()
-        self.assertIn('"generated" in funcName', text)
+        self.assertIn('"generated", line 2, in funcName', text)
```
Comment (felixxm) [inline tests/view_tests/tests/test_debug.py]: We want to change whitespaces in the traceback, so we should test these changes e.g.
```python
        self.assertIn(
            '  File "generated", line 2, in funcName\n'
            '    <source code not available>',
            text,
        )
```

**Comment by jerrodmartin** [comment]
I made the requested changes and updated the HTML template @felixxm

**Comment by felixxm** [comment]
@jerrodmartin Thanks :+1: Welcome aboard :boat:

**Pull request merged**